qiskit_experiments/library/randomized_benchmarking/layer_fidelity_unitary.py

```diff
@@ -0,0 +1,606 @@
+)
+from .layer_fidelity_analysis import LayerFidelityAnalysis
+
+LOG = logging.getLogger(__name__)
+
+
+GATE_NAME_MAP = get_standard_gate_name_mapping()
+NUM_1Q_CLIFFORD = CliffordUtils.NUM_CLIFFORD_1_QUBIT
+
+
+class LayerFidelityUnitary(BaseExperiment):
+    r"""A holistic benchmarking experiment to characterize the full quality of the devices at scale.
+
+    # section: overview
+        Unitary Layer Fidelity (ULF) is a method to estimate the fidelity of
+        a connecting set of arbitrary two-qubit gates over :math:`N` qubits by measuring gate errors
+        using simultaneous direct unitary randomized benchmarking (RB) in disjoint layers.
+        LF can easily be expressed as a layer size independent quantity, error per layered gate (EPLG):
+        :math:`EPLG = 1 - LF^{1/N_{2Q}}` where :math:`N_{2Q}` is number of 2-qubit gates in the layers.
+
+        Each of the 2-qubit (or 1-qubit) direct RBs yields the decaying probabilities
+        to get back to the ground state for an increasing sequence length (i.e. number of layers),
+        fits the exponential curve to estimate the decay rate, and calculates
+        the process fidelity of the subsystem from the rate.
+        LF is calculated as the product of the 2-qubit (or 1-qubit) process fidelities.
+        See Ref. [1] for details.
+
+        This unitary version allows artibrary 2Q gates
+
+    # section: analysis_ref
+        :class:`LayerFidelityAnalysis`
+
+    # section: reference
+        .. ref_arxiv:: 1 2311.05933
+
+    # section: example
+        .. jupyter-execute::
+            :hide-code:
+
+            # backend
+            from qiskit_aer import AerSimulator
+            from qiskit_aer.noise import NoiseModel, depolarizing_error
+
+            noise_model = NoiseModel()
+            noise_model.add_all_qubit_quantum_error(depolarizing_error(5e-3, 1), ["sx", "x"])
+            noise_model.add_all_qubit_quantum_error(depolarizing_error(0, 1), ["rz"])
+            noise_model.add_all_qubit_quantum_error(depolarizing_error(5e-2, 2), ["cx"])
+            backend = AerSimulator(noise_model=noise_model)
+
+        .. jupyter-execute::
+
+            import numpy as np
+            from qiskit import QuantumCircuit
+            from qiskit_experiments.library import StandardRB
+            from qiskit_experiments.library.randomized_benchmarking import LayerFidelityUnitary
+
+            lengths = np.arange(1, 80, 10)
+            two_qubit_layers=[[(0, 1), (3, 5)], [(1, 3), (5, 6)]]
+
+            num_samples = 3
+            seed = 106
+
+            qc = QuantumCircuit(2)
+            qc.rzz(0.5,0,1)
+
+            exp = LayerFidelityUnitary(
+                    physical_qubits=[0, 1, 3, 5, 6],
+                    two_qubit_layers=two_qubit_layers,
+                    lengths=lengths,
+                    backend=backend,
+                    num_samples=num_samples,
+                    seed=seed,
+                    two_qubit_gates=[qc.to_instruction()],
```

Comment:
We could do `from qiskit.circuit.library import RZZGate` and `two_qubit_gates=[RZZGate(0.5)]` here.

Reply:
I guess this was more illustrative of how it would be done in a general case?